build.sbt

```diff
@@ -601,7 +550,19 @@ lazy val spark = (project in file("spark-unified"))
       (sparkV1 / baseDirectory).value / "src" / "test" / "resources"
     ),
 
-    crossSparkSettings(),
+    CrossSparkVersions.sparkDependentSettings(sparkVersion),
+
+    // Override Jackson versions to match Spark version and avoid conflicts with kernel
+    dependencyOverrides ++= {
+      val jacksonVer = CrossSparkVersions.jacksonVersionForSpark(sparkVersion).value
+      Seq(
+        "com.fasterxml.jackson.core" % "jackson-databind" % jacksonVer,
+        "com.fasterxml.jackson.core" % "jackson-core" % jacksonVer,
+        "com.fasterxml.jackson.core" % "jackson-annotations" % jacksonVer,
+        "com.fasterxml.jackson.datatype" % "jackson-datatype-jdk8" % jacksonVer,
+        "com.fasterxml.jackson.module" %% "jackson-module-scala" % jacksonVer
+      )
+    },
```

Comment:
minor - maybe factor this out to a fx or something? see it repeated a lot throughout the build file

Reply:
mind if this is done as a follow up. this PR is already large and at this point i am trying to not boil the ocean and keep my changes minimum :)